src/SIS_continuity.F90

```diff
@@ -120,6 +120,7 @@ subroutine ice_continuity(u, v, hin, h, uh, vh, dt, G, US, IG, CS, use_h_neg, ma
   stensil = 3 ; if (CS%simple_2nd) stensil = 2 ; if (CS%upwind_1st) stensil = 1
 
   do k=1,nCat ; do j=js,je ; do i=is,ie ; if (h(i,j,k) < 0.0) then
+    print *, 'Negative ice thickness at:', i+G%idg_offset, j+G%jdg_offset, k, h(i,j,:)
```

Comment:
We have been encouraged not to write directly to stdout using `write (*,*)` or `print *` to prepare for machines where not all PEs are able to access the I/O channels, and we are not doing so elsewhere in the MOM6 or SIS2 code.

  Please revise these useful messages to write to a message string, and then either concatenate this message string to the following `SIS_error(FATAL, ...)` call, or as 'call SIS_error(WARNING, mesg, all_print=.true.)` before the fatal error message, and similarly for the other 5 new error messages in this file and the other new error message in SIS_transport.F90 using the similar `print *,...` construct.